### ISSUE TITLE
The 'or' expression does not evaluate parameters passed in correctly in certain cases

### ISSUE THREAD

**Issue by ydu34** [MicrosoftDocs/azure-devops-docs#12963]
When using the or expression, it does not evaluate parameters that are passed in correctly in certain cases. 

For example, there are two Boolean parameters:
`parameter1 = False`
`parameter2 = False`

when using 
`or(parameter1, parameter2)`
 it will result in `True`, even though it is expected to result in `False`

I printed the parameters individually and also did comparisons to test
`eq(parameter1, 'false')`
`eq(parameter1, 'False')`
`eq(parameter1, False)`
will all evaluate to True, and 
`eq(parameter1, 'true')`
`eq(parameter1, 'True')`
`eq(parameter1, True)`
will all evaluate to False

so when using 
`or(eq(parameter1, 'true'), eq(parameter2, 'true'))`
it will result with the correct Boolean `False`

---
#### Document Details

⚠ *Do not edit this section. It is required for learn.microsoft.com ➟ GitHub issue linking.*

* ID: 77c58a78-a567-e99a-9eb7-62dddd1b90b6
* Version Independent ID: 680a79bc-11de-39fc-43e3-e07dc762db18
* Content: [Expressions - Azure Pipelines](https://learn.microsoft.com/en-us/azure/devops/pipelines/process/expressions?view=azure-devops)
* Content Source: [docs/pipelines/process/expressions.md](https://github.com/MicrosoftDocs/azure-devops-docs/blob/main/docs/pipelines/process/expressions.md)
* Service: **azure-devops-pipelines**
* Sub-service: **azure-devops-pipelines-process**
* GitHub Login: @juliakm
* Microsoft Alias: **jukullam**

**Comment by WilliamAntonRohm** [comment]
@ydu34 -- Yu, thank you for your research and feedback.  That article states:
- [or](https://learn.microsoft.com/en-us/azure/devops/pipelines/process/expressions?view=azure-devops#or)
> Casts parameters to Boolean for evaluation

@juliakm -- Julia, is the following relevant to Yu's Boolean literals?  That is, are they mistakenly viewed as strings?

- [String](https://learn.microsoft.com/en-us/azure/devops/pipelines/process/expressions?view=azure-devops#string-1)
> To Boolean: '' (the empty string) → False, any other string → True

Any non-empty string, whether 'True' or 'False' or any variation, is cast to a Boolean True.

**Comment by juliakm** [comment]
@WilliamAntonRohm Yes, my guess here is that they are running into the constraint that all variables are really strings. 

@ydu34 Are you working with [parameters](https://learn.microsoft.com/en-us/azure/devops/pipelines/process/templates?view=azure-devops#parameters) or [variables](https://learn.microsoft.com/en-us/azure/devops/pipelines/process/variables?view=azure-devops&tabs=yaml%2Cbatch) in these examples?

**Comment by juliakm** [comment]
Closing issue after no response for 2 weeks.
